src/batch/src/executor/join/lookup_join.rs

```diff
@@ -226,94 +228,128 @@ impl<P: 'static + ProbeSideSourceBuilder> LookupJoinExecutor<P> {
             None.into(),
         ));
 
-        while let Some(data_chunk) = build_side_stream.next().await {
-            let data_chunk = data_chunk?;
+        while let Some(build_chunk) = build_side_stream.next().await {
+            let build_chunk = build_chunk?.compact()?;
+
+            let mut min_heap = self.build_min_heap(&build_chunk)?;
+
+            while min_heap.peek().is_some() {
```

Comment:
Maybe the min_heap is unnecessary. You actually want group by key instead of sorting? So what about

```rust
        while let Some(build_chunk) = build_side_stream.next().await {
            let groups = build_chunk?.rows().group_by(|row| {
                self.build_side_key_idxs
                    .iter()
                    .map(|&idx| row.value_at(idx).to_owned_datum().unwrap())
                    .collect_vec()
            });

            for (key, group) in &groups {
                for cur_row in group {
```

Reply:
And to implement further grouping, you just need to group by vnode

Reply:
Oops this approach doesn't work due to the implementation details of itertools::GroupBy 😅
<img width="1154" alt="image" src="https://user-images.githubusercontent.com/37948597/180237886-50963463-5f20-4155-abd8-e4622d17b383.png">

Reply:
It can work by replacing `group_by` to `into_group_map_by` 😇

Reply:
Thanks for the suggestion!